activerecord/lib/rails/generators/active_record/model/model_generator.rb

```diff
@@ -43,9 +43,16 @@ def accessible_attributes
 
         # Used by the migration template to determine the parent name of the model
         def parent_class_name
-          options[:parent] || "ActiveRecord::Base"
+          options[:parent] || determine_default_parent_class
         end
 
+        def determine_default_parent_class
+          if File.exist?('app/models/application_record.rb')
+            "ApplicationRecord"
+          else
+            "ActiveRecord::Base"
+          end
+        end
```

Comment:
Did that mainly for apps upgrading to Rails 5. On the one hand this may push people into using `ApplicationRecord` which may have ignored it before, but on the other one – should we be that defensive?